aidb/engine/approx_aggregate_join_engine.py

```diff
@@ -0,0 +1,288 @@
+from typing import Dict, List
+
+import numpy as np
+import pandas as pd
+import sqlglot.expressions as exp
+from sqlalchemy.sql import text
+
+from aidb.engine.approx_aggregate_engine import ApproximateAggregateEngine
+from aidb.query.query import Query
+from aidb.utils.constants import (MASS_COL_NAME, NUM_ITEMS_COL_NAME,
+                                  WEIGHT_COL_NAME)
+from aidb.utils.logger import logger
+
+_NUM_PILOT_SAMPLES = 3000000
```

Comment:
Are we using a pilot sample size of 3 million?

Reply:
For a pilot sample size of 1 million, approximately 46 million samples are needed to achieve the desired results. When the size increases to 2 million, the required number decreases to 37 million. Further increasing the size to 3 million reduces the required number to 21 million.